pkg/cloud/services/instancestate/rule.go

```diff
@@ -0,0 +1,214 @@
+	})
+
+	if err != nil {
+		return errors.Wrap(err, "unable to get queue attributes")
+	}
+
+	targetsResp, err := s.EventBridgeClient.ListTargetsByRule(&eventbridge.ListTargetsByRuleInput{
+		Rule: aws.String(s.getEC2RuleName()),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "unable to list targets for rule %s", s.getEC2RuleName())
+	}
+
+	targetFound := false
+	for _, target := range targetsResp.Targets {
+		// check if queue is already added as a target
+		if *target.Id == GenerateQueueName(s.scope.Name()) && *target.Arn == *queueAttrs.Attributes[sqs.QueueAttributeNameQueueArn] {
+			targetFound = true
+		}
+	}
+
+	if !targetFound {
+		_, err = s.EventBridgeClient.PutTargets(&eventbridge.PutTargetsInput{
+			Rule: ruleResp.Name,
+			Targets: []*eventbridge.Target{{
+				Arn: queueAttrs.Attributes[sqs.QueueAttributeNameQueueArn],
+				Id:  aws.String(GenerateQueueName(s.scope.Name())),
+			}},
+		})
+
+		if err != nil {
+			return errors.Wrapf(err, "unable to add SQS target %s to rule %s", GenerateQueueName(s.scope.Name()), s.getEC2RuleName())
+		}
+	}
+
+	if queueAttrs.Attributes[sqs.QueueAttributeNamePolicy] == nil {
+		// add a policy for the rule so the rule is authorized to emit messages to the queue
+		err = s.createPolicyForRule(&createPolicyForRuleInput{
+			QueueArn: *queueAttrs.Attributes[sqs.QueueAttributeNameQueueArn],
+			QueueURL: *queueURLResp.QueueUrl,
+			RuleArn:  *ruleResp.Arn,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s Service) createRule() error {
+	eventPattern := eventPattern{
+		Source:     []string{"aws.ec2"},
+		DetailType: []string{Ec2StateChangeNotification},
+		EventDetail: &eventDetail{
+			States: []infrav1.InstanceState{infrav1.InstanceStateShuttingDown, infrav1.InstanceStateTerminated},
+		},
+	}
+	data, _ := json.Marshal(eventPattern)
+	// create in disabled state so the rule doesn't pick up all EC2 instances. As machines get created,
+	// the rule will get updated to track those machines
+	_, err := s.EventBridgeClient.PutRule(&eventbridge.PutRuleInput{
+		Name:         aws.String(s.getEC2RuleName()),
+		EventPattern: aws.String(string(data)),
+		State:        aws.String(eventbridge.RuleStateDisabled),
+	})
+
+	return err
+}
+
+func (s Service) deleteRules() error {
+	_, err := s.EventBridgeClient.RemoveTargets(&eventbridge.RemoveTargetsInput{
+		Rule: aws.String(s.getEC2RuleName()),
+		Ids:  aws.StringSlice([]string{GenerateQueueName(s.scope.Name())}),
+	})
+	if err != nil && !resourceNotFoundError(err) {
+		return errors.Wrapf(err, "unable to remove target %s for rule %s", GenerateQueueName(s.scope.Name()), s.getEC2RuleName())
+	}
+	_, err = s.EventBridgeClient.DeleteRule(&eventbridge.DeleteRuleInput{
+		Name: aws.String(s.getEC2RuleName()),
+	})
+
+	if err != nil && resourceNotFoundError(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "unable to delete rule %s", s.getEC2RuleName())
+}
+
+func (s Service) AddInstanceToEventPattern(instanceID string) error {
```

Comment:
I see deleted instances still exist on the rule, we may need to have a cleanup for this during reconcileDelete.

Reply:
I had this on my initial plan but seemed to have skipped it as failed/terminated instances won't be emitting any new events. Will update to clean up the rule before an intentional deletion of the machine.